docs/content/how-to/aws/create-aws-hosted-cluster-autonode.md

```diff
@@ -0,0 +1,300 @@
+
+```sh
+cat << EOF | oc apply -f -
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+ name: web-app
+spec:
+ replicas: 0
+ selector:
+   matchLabels:
+     app: web-app
+ template:
+   metadata:
+     labels:
+       app: web-app
+   spec:
+     affinity:
+       podAntiAffinity:
+         requiredDuringSchedulingIgnoredDuringExecution:
+           - labelSelector:
+               matchLabels:
+                 app: web-app
+             topologyKey: "kubernetes.io/hostname"   
+     securityContext:
+       runAsUser: 1000
+       runAsGroup: 3000
+       fsGroup: 2000
+     containers:
+     - image: public.ecr.aws/eks-distro/kubernetes/pause:3.2
+       name: web-app
+       resources:
+         requests:
+           cpu: "1"
+           memory: 256M
+       securityContext:
+         allowPrivilegeEscalation: false
+     nodeSelector:
+       node.kubernetes.io/instance-type: "t3.large"
+EOF
+```
+
+Scale the application:
+
+```sh
+oc scale --replicas=1 deployment.apps/web-app
+```
+
+**Monitor and Debug Node Provisioning with AutoNode:**
+
+Monitor the `nodeClaims` objects to track the provisioning of nodes by AutoNode. These objects provide detailed insights into the lifecycle of nodes, including their current state and any associated events. Use the following command to continuously watch the `nodeClaims`:
+
+!!! warning "Provisioning Time"
+  The provisioning process for an instance to become a node may take approximately 10 minutes. While waiting, monitor the progress using the provided commands to ensure the process completes successfully.
+
+
+```sh
+oc get nodeclaims --watch
+```
+
+To investigate a specific `nodeClaim` in detail, use the following command:
+
+```sh
+oc describe nodeclaim <nodeClaimName>
+```
+
+This will provide comprehensive information about the selected `nodeClaim`, helping you debug and confirm that nodes are being provisioned and functioning as expected.
+
+
+**Verify Node Join Status:**
+
+Ensure that the node has successfully joined the cluster. Use the following command to check:
+
+```sh
+oc get nodes -l karpenter.sh/nodepool=spot-and-gpu
+```
+
+This command filters the nodes associated with the `spot-and-gpu` NodePool, allowing you to confirm that the AutoNode feature is functioning as expected.
+
+**Verify Application Scheduling:**
+
+Ensure that the application has been successfully scheduled onto the newly provisioned node. Use the following command to check the status of the pods associated with the application:
+
+```sh
+oc get pods -l app=web-app -o wide
+```
+
+This command filters the pods by the label `app=web-app`, allowing you to confirm that the application is running on the expected node provisioned by AutoNode.
+
+
+[featuregate]: https://hypershift-docs.netlify.app/how-to/feature-gates
```

Comment:
Let's add a note, here "openshiftNodeClass is exposed for API consumers https://github.com/openshift/hypershift/blob/main/api/karpenter/v1beta1/karpenter_types.go#L53"